components/provisioning/org.wso2.carbon.identity.provisioning/src/test/java/org/wso2/carbon/identity/provisioning/ProvisioningUtilTest.java

```diff
@@ -0,0 +1,385 @@
+        assertNull(value2);
+    }
+
+
+    @Test(dataProvider = "getInboundClaimData")
+    public void testGetMappedClaims(String outboundClaimDialect,
+                                    Object inboundClaimValueMap, Object inboundClaimMappings,
+                                    Object outboundClaimValueMappings, String tenantDomain) throws Exception {
+
+        Map<ClaimMapping, List<String>> result = null;
+        result = ProvisioningUtil.getMappedClaims(outboundClaimDialect,
+                (Map<String, String>) inboundClaimValueMap, (ClaimMapping[]) inboundClaimMappings,
+                (Map<ClaimMapping, List<String>>)
+                        outboundClaimValueMappings, tenantDomain);
+        assertNotNull(result);
+    }
+
+    @Test(dataProvider = "getInboundClaimData1")
+    public void testGetMappedClaims(String outboundClaimDialect,
+                                    Object inboundClaimValueMap, String inboundClaimMappingDialect,
+                                    Object outboundClaimValueMappings, String tenantDomain)
+            throws Exception {
+
+
+        Map claimValueMap = (Map<String, String>) inboundClaimValueMap;
+        Set keySet = null;
+        if (MapUtils.isNotEmpty(claimValueMap)) {
+            keySet = claimValueMap.keySet();
+        }
+
+        Map<String, String> claimMappings = new HashMap<>();
+
+        claimMappings.put(localClaimUri, "testOutboundUser");
+        claimMappings.put(localClaimUri + "1", "testOutboundUser1");
+        claimMappings.put("test", localClaimUri);
+
+        when(mockClaimMetadataHandler.getMappingsMapFromOtherDialectToCarbon(inboundClaimDialect,
+                keySet, tenantDomain, false)).thenReturn(claimMappings);
+        Map<ClaimMapping, List<String>> result = null;
+        try {
+            result = ProvisioningUtil.getMappedClaims(outboundClaimDialect,
+                    claimValueMap, inboundClaimMappingDialect, (Map<ClaimMapping,
+                            List<String>>)
+                            outboundClaimValueMappings, tenantDomain);
+            assertNotNull(result);
+
+        } catch (IdentityApplicationManagementException e) {
+
+        }
+    }
+
+    @Test(dataProvider = "getInboundClaimData2")
+    public void testGetMappedClaims(Object outboundClaimMappings, Object inboundClaimValueMap,
+                                    Object inboundClaimMappings,
+                                    Object outboundClaimValueMappings) {
+
+        Map<ClaimMapping, List<String>> result = null;
+        try {
+            result = ProvisioningUtil.getMappedClaims((ClaimMapping[]) outboundClaimMappings,
+                    (Map<String, String>) inboundClaimValueMap, (ClaimMapping[]) inboundClaimMappings,
+                    (Map<ClaimMapping,
+                            List<String>>) outboundClaimValueMappings);
+            assertNotNull(result);
+
+        } catch (IdentityApplicationManagementException e) {
```

Comment:
Any reason why you catch this exception ? Usually we dont handle them at tests. Just allowing them to cause so that the test will fail.